### DISCUSSION TITLE
cURL manifest updates

### DISCUSSION THREAD

**Discussion by szhu25** [microsoft/winget-pkgs#122883] [Polls]
Hi Winget community,

While cURL version is currently staying at 8.4.0, cURL team has released 3 different builds (8.4.0_1, 8.4.0_2, 8.4.0_3) for now and the current build winget includes is 8.4.0_1. Based on [this blog](https://daniel.haxx.se/blog/2018/12/19/why-is-curl-different-everywhere/) and my monitoring software, it looks like the difference primarily is the library used in the build process. 
On the last cURL version, I've been trying to keep up with the new builds cURL releases and submit a PR for each build (same manifest, modified file url). I have not done so in this version.  
cURL could release more than one builds per day, and as of the current setup for winget I do not believe winget would tell users to update to the new build if they have a older one installed. Would it be worth it to still keep up with the build changes? For example, update the manifests immediately? Daily? Weekly? 

P.S. I also noticed that Windows would still use the default cURL version shipped with the OS release in cmd, so I'm not even sure keeping cURL binary updated on winget would help anyhow... 

Thanks!

**Comment by mdanish-kh** [comment]
> I do not believe winget would tell users to update to the new build if they have a older one installed.

Why is that? Since cURL is a portable package, we actually control what version is written to Control Panel / Registry in the manifest. We can have versions like 8.4.0.1, 8.4.0.2, 8.4.0.3 in the repo and I imagine WinGet will prompt to update to a newer build version if it's present in the repo. (and also if the previous version was one that was installed through WinGet)

> Would it be worth it to still keep up with the build changes? For example, update the manifests immediately? Daily? Weekly?

My answer to this would be that's up to the contributors/users of the package to decide. If they want a specific build version, they can request for it in the issues or add a PR for it. I personally feel it's daunting to keep up with build changes without automation.

**Reply by szhu25** [reply]
I was thinking it's hard to force winget to update without changing the version number structure. Adding a build number at the end is probably a good idea. 
I think Komac and others are probably capable of automating the build submission once someone is able to figure out how to parse and change the folder path inside archive (version_build). 

I'll submit a PR to Komac and see if it's something others already have knowledge on.

**Comment by szhu25** [comment]
Current changes:
1. Build numbers are added at the end of normal cURL version number
2. I'll check and submit manifest twice per week. Normally it would be Wednesday & weekends. Unless a version change is done (which would be ASAP on my schedule)

**Comment by ThorstenGiesecke** [comment]
Beside that, winget offers an outdated 8.13 build. Microsoft's version of curl omits several features, including HTTP/2 and HTTP/3 support, Public Suffix List (PSL) support, Gopher, LDAP, and Brotli compression. So options like --http2 or --http2-prior-knowledge are not available and cookies can be shared across different registered domains due lack of PSL.

**Reply by szhu25** [reply]
Actually, the latest cURL version available on Winget is 8.14.0 (Build 1), which aligns with the current latest release from the official cURL website (curl.se). The manifest update process for Winget is largely automated, so the cURL package generally stays current. You can see the Winget package details here: https://github.com/microsoft/winget-pkgs/tree/master/manifests/c/cURL/cURL
